### PR TITLE
In-game loadout creation and editing

### PR DESCRIPTION
**Pull request by bhollis**
This moves our ingame loadout state into the Redux store, so we can do updates without having to refresh the profile. It also adds a loadout creator/editor that allows selecting the name, icon, and color, and the slot when creating a new loadout.

Screenshot 2023-03-27 at 12 27 16 AM
Screenshot 2023-03-27 at 12 27 25 AM

### CODE REVIEW

**Comment by nev-r** [comment]
- "snapshot" seems like an entirely bungie-internal term, and is probably not adequate label for that button
- prefer a more detailed label for the "edit" button since all we can edit is the identifiers
- inventory refresh automatically closes the snapshot sheet, it's a little annoying

**Comment by nev-r** [comment]
> Agreed. I'm not sure how to phrase all of these to distinguish between DIM's loadouts and the ingame ones, especially as we add functionality to transfer between them.

we'll need to distinguish a lot less when (if) we displaying them as full loadouts on this page

**Comment by bhollis** [comment]
> inventory refresh automatically closes the snapshot sheet, it's a little annoying

Yeah that's annoying. It's because the current loadout has a new ID every time, which is really useful for being able to save it without special-casing. I'll probably end up having to move this sheet out to a top level component like how LoadoutDrawer works.

File: config/i18n.json [hunk 1]
```diff
@@ -395,8 +395,18 @@
     "TransferItems": "Transfer"
   },
   "InGameLoadout": {
+    "Create": "Create Loadout",
+    "CreateTitle": "Snapshot Equipped to Loadout",
```
Comment (nev-r) [inline config/i18n.json:399]:
```suggestion
    "CreateTitle": "Create In-Game Loadout From Current Equipment",
```

File: config/i18n.json [hunk 2]
```diff
@@ -700,6 +710,7 @@
       "Empty": "The loadout is empty.",
       "NoName": "The loadout needs a name."
     },
+    "Snapshot": "Snapshot",
```
Comment (nev-r) [inline config/i18n.json:713]:
```suggestion
    "Snapshot": "Save As In-Game Loadout",
```

File: src/locale/en.json [hunk 3]
```diff
@@ -726,6 +736,7 @@
     },
     "ShareLoadout": "Share Loadout",
     "ShowModPlacement": "Show Mod Placement",
+    "Snapshot": "Snapshot",
```
Comment (nev-r) [inline src/locale/en.json:739]:
```suggestion
    "Snapshot": "Save As In-Game Loadout",
```

File: src/locale/en.json [hunk 4]
```diff
@@ -387,7 +387,17 @@
   },
   "InGameLoadout": {
     "Applying": "Equipping loadout...",
-    "LoadoutSlotNum": "Slot {{index}}"
+    "Create": "Create Loadout",
+    "CreateTitle": "Snapshot Equipped to Loadout",
```
Comment (nev-r) [inline src/locale/en.json:391]:
```suggestion
    "CreateTitle": "Create In-Game Loadout From Current Equipment",
```

**Review by nev-r — APPROVED** [review]

**Pull request merged**